File: bloomstack_core/public/js/account_opened_date.js

frappe.ui.form.on('Lead', {
    onload: function(frm) {
        if(frm.doc.status == 'Converted') {
            frappe.db.get_value("Customer", {"lead_name": frm.doc.name}, "creation", (r) => {
				if(r) {
					frm.set_value("account_opened_date", r.creation);
				}	
			});

      }

    }

});

File: bloomstack_core/patches/v0_0_1/account_opened_date.py
import frappe
from frappe.modules.utils import sync_customizations


def execute():
	sync_customizations("bloomstack_core")

	leads = frappe.get_all("Lead", fields=["name", "creation"])
	for lead in leads:
		result = frappe.db.get_value("Customer", {"lead_name": lead.name}, ["creation", "opening_date"], as_dict=True)
		if result:
			frappe.db.set_value("Lead", lead.name, "account_opened_date", result.opening_date or result.creation)

	customers = frappe.get_all("Customer", filters={"lead_name": ("!=", "")}, fields=["opening_date", "lead_name"])
	for customer in customers:
		if customer.lead_name:
			frappe.db.set_value("Lead", customer.lead_name, "account_opened_date", customer.opening_date)
